ProductionSystem/DB/ProductionDB.py

```diff
@@ -0,0 +1,532 @@
+""" DIRAC Production DB
+
+    Production database is used to collect and serve the necessary information
+    in order to automate the task of transformation preparation for high level productions.
+"""
+
+# # imports
+import json
+import threading
+
+from DIRAC import gLogger, S_OK, S_ERROR
+from DIRAC.Core.Base.DB import DB
+from DIRAC.ProductionSystem.Utilities.ProdValidator import ProdValidator
+from DIRAC.ProductionSystem.Utilities.ProdTransManager import ProdTransManager
+from DIRAC.Core.Utilities.List import intListToString
+
+
+__RCSID__ = "$Id$"
+
+MAX_ERROR_COUNT = 10
+
+#############################################################################
+
+
+class ProductionDB(DB):
+  """ ProductionDB class
+  """
+
+  def __init__(self, dbname=None, dbconfig=None, dbIn=None):
+    """ The standard constructor takes the database name (dbname) and the name of the
+        configuration section (dbconfig)
+    """
+    if not dbname:
+      dbname = 'ProductionDB'
+    if not dbconfig:
+      dbconfig = 'Production/ProductionDB'
+
+    if not dbIn:
+      DB.__init__(self, dbname, dbconfig)
+
+    self.lock = threading.Lock()
+
+    self.prodValidator = ProdValidator()
+    self.ProdTransManager = ProdTransManager()
+
+    self.PRODPARAMS = ['ProductionID',
+                       'ProductionName',
+                       'Description',
+                       'CreationDate',
+                       'LastUpdate',
+                       'AuthorDN',
+                       'AuthorGroup',
+                       'Status']
+
+    self.TRANSPARAMS = ['TransformationID',
+                        'ProductionID',
+                        'LastUpdate',
+                        'InsertedTime']
+
+    self.TRANSLINKSPARAMS = ['TransformationID',
+                             'ParentTransformationID',
+                             'ProductionID']
+
+    self.statusActionDict = {
+        'New': None,
+        'Active': 'startTransformation',
+        'Stopped': 'stopTransformation',
+        'Cleaned': 'cleanTransformation'}
+
+  def addProduction(self, prodName, prodDescription, authorDN, authorGroup, connection=False):
+    """ Create new production starting from its description
+
+    :param prodName: a string with the Production name
+    :param prodDescription: a json object with the Production description
+    :param authorDN: string with the author DN
+    :param authorGroup: string with author group
+    """
+    connection = self.__getConnection(connection)
+    res = self._getProductionID(prodName, connection=connection)
+    if res['OK']:
+      return S_ERROR("Production with name %s already exists with ProductionID = %d" % (prodName,
+                                                                                        res['Value']))
+    elif res['Message'] != "Production does not exist":
+      return res
+
+    self.lock.acquire()
+    req = "INSERT INTO Productions (ProductionName, Description, CreationDate,LastUpdate, \
```

Comment:
The newly inserted values are not escaped - beware SQL injection !

Reply:
Could you please more explicit?

Reply:
Sorry, I was not clear. I mean that newly inserted strings are passed directly from the client without "escaping" them for protection against SQL injection. The string values should be processed before inserting them, e.g.:

   result = self._escapeString(prodName):
   if not result['OK']:
     return result
   prodName = result['Value']

And the same for prodDescription